packages/core/src/scripts/migrate-oz-cli-project.ts

```diff
@@ -0,0 +1,426 @@
+          label: stripContractName(member.label),
+          type: transformTypeName(member.type),
+        })),
+      };
+    case 'Enum':
+      return {
+        label: stripContractName(oldType.label),
+        members: oldType.members,
+      };
+    default:
+      return {
+        label: oldType.label,
+      };
+  }
+}
+
+function transformTypeName(typeName: string): string {
+  switch (getTypeKind(typeName)) {
+    case 'Struct':
+      return transformStructTypeName(typeName);
+    case 'Enum':
+      return transformEnumTypeName(typeName);
+    case 'Mapping':
+      return transformMappingTypeName(typeName);
+    case 'DynArray':
+      return transformDynArrayTypeName(typeName);
+    case 'StaticArray':
+      return transformStaticArrayTypeName(typeName);
+    case 'Elementary':
+      return typeName;
+    default:
+      throw new Error(`Unknown type: ${typeName}`);
+  }
+}
+
+function transformStructTypeName(typeName: string): string {
+  const valueType = stripContractName(getValueType(typeName));
+  return `t_struct(${valueType})_storage`;
+}
+
+function transformEnumTypeName(typeName: string): string {
+  const valueType = stripContractName(getValueType(typeName));
+  return `t_enum(${valueType})`;
+}
+
+function transformMappingTypeName(typeName: string): string {
+  const valueType = transformTypeName(getValueType(typeName));
+  return `t_mapping(${valueType})`;
+}
+
+function transformDynArrayTypeName(typeName: string): string {
+  const valueType = transformTypeName(getValueType(typeName));
+  return `t_array(${valueType})dyn_storage`;
+}
+
+function transformStaticArrayTypeName(typeName: string): string {
+  const size = optimisticMatch(typeName, /:(\d+)/)[1];
+  const valueType = transformTypeName(getValueType(typeName));
+  return `t_array(${valueType})${size}_storage`;
+}
+
+function getValueType(typeName: string): string {
+  return optimisticMatch(typeName, /<(.+)>/)[1];
+}
+
+function stripContractName(s: string): string {
+  return optimisticMatch(s, /(.+)\.(.+)/)[2];
+}
+
+function optimisticMatch(s: string, rg: RegExp): string[] {
+  const matches = s.match(rg);
+  if (matches === null) {
+    return [s];
+  }
+  return matches;
+}
+
+function getTypeKind(typeName: string): TypeKind {
+  if (/^t_struct<.+>/.test(typeName)) {
+    return 'Struct';
+  } else if (/^t_enum<.+>/.test(typeName)) {
+    return 'Enum';
+  } else if (/^t_mapping<.+>/.test(typeName)) {
+    return 'Mapping';
+  } else if (/^t_array:dyn<.+>/.test(typeName)) {
+    return 'DynArray';
+  } else if (/^t_array:\d+<.+>/.test(typeName)) {
+    return 'StaticArray';
+  } else {
+    return 'Elementary';
+  }
+}
+
+type TypeKind = 'Elementary' | 'Mapping' | 'Struct' | 'Enum' | 'DynArray' | 'StaticArray';
+
+interface MigrationOutput {
+  newManifests: Record<string, ManifestData>;
+  manifestsMigrationData: unknown;
+}
+
+interface AddressWrapper {
+  address?: string;
+}
+
+interface ExportData {
```

Comment:
I was surprised to see so many things are exported, but I think it does make sense to export those that are addresses.

But we should probably delete `manifestVersion`, `zosversion`, `version`, `frozen`, and `dependencies`.

Reply:
The current script doesn't allow `solidityLibs`. The fields in this type are actually the ones in `LegacyManifest` minus the ones we delete from it.

Reply:
Good point, I forgot about that.